### DISCUSSION TITLE
Question about children & contexts

### DISCUSSION THREAD

**Discussion by PubliAlex** [OFFLINE-GmbH/oc-boxes-support#141] [Help]
Hello,

I'm in a situation where I have some boxes that can be added everywhere (ex : row or column box) and some images that can only be child of row or column box.

For that I use the children and contexts options : 

    name: Row
    handle: layout\row
    section: Structure
    
    children: 
        - structure
        - basic

    contexts:
        - structure
---
    name: Column
    handle: layout\col
    section: Structure
    
    children:
        - structure
        - basic
    
    contexts:
        - structure
---
    name: Text
    handle: common\text
    section: Basic
    
    contexts:
        - basic

My problem here is that I want that column or row box be added in the page root, which is disabled because I defined a context. But I also want that row or column be a child of other row or column, so I'm obliged to define children and contexts.

To me, there is a missing "root" context to allow that, I don't know if it's already available, the documentation don't mention something like that. How could I solve this problem ?

Thank you

**Comment by tobias-kuendig** [comment]
@PubliAlex does using the `default` context work for you?

https://docs.boxes.offline.ch/reference/schema.html#contexts

So that would be:

```yaml
name: Column
handle: layout\col
section: Structure

children:
    - structure
    - basic

contexts:
    - structure
    - default # add this
 ```
